### ISSUE TITLE
Integrated Terminal (PowerShell) - The handle is invalid

### ISSUE THREAD

**Issue by ljt91** [Microsoft/vscode#10625]
- VSCode Version: 1.4.0
- OS Version: Windows 10 Enterprise 1511

Steps to Reproduce:
Required PowerShell Module: AzureRm
1. View -> Integrated Terminal (PowerShell) -> Login-AzureRmAccount

Error:

```
Login-AzureRmAccount : The handle is invalid
At line:1 char:1
+ Login-AzureRmAccount
+ ~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [Add-AzureRmAccount], AadAuthenticationFailedException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.Profile.AddAzureRMAccountCommand
```

Thanks,
Lyon Till

**Comment by dawilliams** [comment]
I just had this happen to me as well.

David

**Comment by Tyriar** [comment]
Are you using 32 bit powershell (In C:\Windows\System32) as opposed to 64-bit? See https://code.visualstudio.com/docs/editor/integrated-terminal#_configuration

Let me know if that doesn't fix it.

**Comment by dawilliams** [comment]
Tyriar,

Mine is set up as follows and the error is still there.

`"terminal.integrated.shell.windows": "C:\\WINDOWS\\sysnative\\WindowsPowerShell\\v1.0\\powershell.exe"`

OS: Windows 10
VSCode: 1.4.0
PS: 5

**Comment by Tyriar** [comment]
@daviwil do you know what the issue could could be here?

**Comment by daviwil** [comment]
Have not seen that before, it might be something that the Azure module is doing that isn't working so well with the integrated console.

**Comment by daviwil** [comment]
Though I have seen people report recently that this cmdlet has problems running using the internal PS session in my language/debug servers, so maybe there's a bug in that module?

**Comment by smneal** [comment]
I've just started trying to use this and hit the same issue.

**Comment by justinbarias** [comment]
Any resolution to this?

**Comment by Structed** [comment]
Same with PowerShell x64

VS Code: 1.5.3
PowerShell: 5.1 Build 14393 Rev 206
Windows 10 Pro 1607, Build 14393.222

Expected Result:
Should open a Authentication Window for Microsoft Account

**Comment by daviwil** [comment]
I'll take a look at this again, @dfinke just reported the same thing to me.

**Comment by vicioussn** [comment]
I've encountered the same issue.

**Comment by olandese** [comment]
Same issue for me too

**Comment by keeshuizer** [comment]
I also ran into this problem, is there any news about a fix or workaround for this?

**Comment by fitzgeraldsteele** [comment]
I also hit this problem

**Comment by the-ress** [comment]
I can reproduce it in 1.8.1, but it's working in the insider build.

I think it was fixed with winpty upgrade in #13625. The cmdlet calls `GetConsoleWindow()` to get owner window for the login dialog. The old winpty probably did something to the console window so `GetConsoleWindow` returned an invalid handle.

**Comment by Tyriar** [comment]
Great :)
